batch_requests.go

```diff
@@ -0,0 +1,150 @@
+package msgraphgocore
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/google/uuid"
+	abstractions "github.com/microsoft/kiota-abstractions-go"
+	"github.com/microsoftgraph/msgraph-sdk-go-core/internal"
+)
+
+// SendBatch sends a batch request
+func SendBatch(batch batchRequest, adapter abstractions.RequestAdapter) (BatchResponse, error) {
+	var res BatchResponse
+
+	batchJsonBody, err := batch.toJson()
+	if weGotAnError(err) {
+		return res, err
+	}
+
+	baseUrl, err := getBaseUrl(adapter)
+	if weGotAnError(err) {
+		return res, err
+	}
+
+	requestInfo := buildRequestInfo(batchJsonBody, baseUrl)
+	return sendBatchRequest(requestInfo, adapter)
+}
+
+// AppendBatchItem converts RequestInformation to a BatchItem and adds it to a BatchRequest
+//
+// You can add upto 20 BatchItems to a BatchRequest
+func (br *batchRequest) AppendBatchItem(reqInfo abstractions.RequestInformation) (*batchItem, error) {
+	if len(br.Requests) > 19 {
+		return nil, errors.New("Batch items limit exceeded. BatchRequest has a limit of 20 batch items")
+	}
+
+	batchItem, err := toBatchItem(reqInfo)
+	if weGotAnError(err) {
+		return nil, err
+	}
+
+	br.Requests = append(br.Requests, batchItem)
+	return batchItem, nil
+}
+
+// DependsOnItem creates a dependency chain between BatchItems.If A depends on B, then B will be sent before B
+func (bi *batchItem) DependsOnItem(item batchItem) {
+	// DependsOn is a single value slice
+	bi.DependsOn = []string{item.Id}
+}
+
+// NewBatchRequest creates an instance of BatchRequest
+func NewBatchRequest() *batchRequest {
+	return &batchRequest{}
+}
+
+func toBatchItem(requestInfo abstractions.RequestInformation) (*batchItem, error) {
+	url, err := requestInfo.GetUri()
```

Comment:
the request information probably won't have the base url set, which means this call will fail. You might need to get the baseUrl from the request adapter and set it on the request information first.

Reply:
The request information is always "missing" the base URL up until it gets executed by the request adapter which provides the value before converting the request info in a native request.
There could be cases where people have set the base URL manually on the object, but I think it'll be much rarer than having set the URL all together (which bypasses the templating mechanism)